quantum/command.c

```diff
@@ -243,7 +243,11 @@ static void print_status(void) {
 
 #if !defined(NO_PRINT) && !defined(USER_PRINT)
 static void print_eeconfig(void) {
+#if defined(LAYER_STATE_32BIT)
+    xprintf("eeconfig:\ndefault_layer: %lu\n", eeconfig_read_default_layer());
+#else
     xprintf("eeconfig:\ndefault_layer: %u\n", eeconfig_read_default_layer());
+#endif
```

Comment:
Currently, the default layer is stored only in an 8-bit variable so I'd like to limit to that, for consistency rather than adding a preprocessor here. 
```suggestion
    xprintf("eeconfig:\ndefault_layer: %u\n", (uint8_t)eeconfig_read_default_layer());
```
Though, I'm not sure that this is the best way to go about this.

(also, noticed that action_layer.c has a couple of debug prints that also error out, if you define `ACTION_DEBUG` with the 32 bit define)

Reply:
`%u` on AVR and ARM are likely two different bitwidths.
The resulting `layer_state_t` can also be a single layer value converted to full-width bitmask (assuming `DEFAULT_LAYER_STATE_IS_VALUE_NOT_BITMASK`), so downcasting to uint8_t is incorrect here.

Suggest using `%lu` unconditionally, and casting to `(long int)` to ensure it matches both AVR and ARM expectations.

Reply:
Thank you for the assist. It works well for me, I've updated the PR with your suggestion